### ISSUE TITLE
DisplayMagician does not start *minimised* and always shows *splash screen*

### ISSUE THREAD

**Issue by Sikahalv** [terrymacdonald/DisplayMagician#111]
DisplayMagician does not start *minimised* and always shows *splash screen*

When you select the following settings below...

* Enable -- "Start DisplayMagician minimised in notification area when computer starts"
* Disable -- "Show DisplayMagician Splash Screen on startup"

DisplayMagician forgets these settings **EVERY REBOOT / LOG OUT** of Windows 10

Very annoying as I don't want to see the splash screen on boot and I want the app to start *minimised* / *hidden*.

~ Windows 10 Pro 21H2
~ DisplayMagician v2.3.1.60

![1](https://user-images.githubusercontent.com/96361352/169627063-4dc87969-e2ed-4955-8d51-1f8675e62aaf.PNG)
![2](https://user-images.githubusercontent.com/96361352/169627065-9193dbd0-0ecb-40a6-80c2-bf8e26215ce9.PNG)

**Comment by terrymacdonald** [comment]
Hi,

How were you setting this? Did you do it in the settings window, or by
checking the checkbox on the main window?

That will help me understand what's broken.

Thanks
Terry

**Comment by Sikahalv** [comment]
Hi Terry,

I've changed this setting either in the main window... or in the settings window... or both... and restarted the computer and neither of those 3 ways of changing the settings actually sticks. It forgets it every reboot or logout. This issue wasn't happening before this major update where my existing profiles were deleted and I had to make new ones.

Regards,
George

**Comment by terrymacdonald** [comment]
Hmmmm, ok.

Can you please generate a Support ZIP File for me (in settings window) and
upload it here through your web browser? I think I need to see what your
settings are in the file.

Thanks
Terry

**Comment by Sikahalv** [comment]
Hi Terry,

Please see attached :)

[DisplayMagician-Support-20220523-1117.zip](https://github.com/terrymacdonald/DisplayMagician/files/8750746/DisplayMagician-Support-20220523-1117.zip)

**Comment by terrymacdonald** [comment]
Hi @Sikalv,

Looking at your settings file, it is set properly. This means that
DisplayMagician is remembering the settings, just not actually running them.

````
 "StartOnBootUp": true,
  "ShowSplashScreen": false,
  "MinimiseOnStart": true,
````
It is set to attempt to start up on boot, and to minimise on start, and
to not show the splashscreen. This is what you said you wanted. Which means
that it is not properly performing this step. This fact is very weird as I
run DisplayMagician v2.3.1.60 myself, and it starts and minimises each time
fine (I don't have ShowSpashScreen disabled myself).

Can you please try and turn the "ShowSplashScreen" back on, and then
restart your computer to see if that fixes it? It may be a logic error. I
will try getting my settings the same as you have selected to see if that
causes any issues on my test machine if I get some time tonight.

I've also looked at your DisplayMagician.log files and they all look fine
to me. There isn't anything particularly odd there. I'm not sure how much
logging I currently do around the enabling load on boot.

Some questions I have after looking at your original notes again:

   - Does DisplayMagician start at all originally after a computer restart
   at all?
   - If it does start originally, does it start minimised?
   - If you start DisplayMagician manually, does it start minimised?
   - If you start DisplayMagician manually, does it show the splash screen?
   - If you go into your Settings window, is the *Start DisplayMagician
   automatically when the computer starts *option currently enabled?
   - If you go into your Settings window, is the *Start DisplayMagician
   minimised in notification area *option currently enabled?
   - Do you run any anti-virus or other start menu adjusting software? Or
   any sort of software that prevents boot up of software?

And lastly, can you please try unchecking the checkbox on the
DisplayMagician Main Window, then rechecking it again, and then restart
DisplayMagician manually and tell me what happens please?

* All of the above steps and questions are designed to help me figure out
where the problem may lie in the code. It may seem like you're answering
the exact same thing over and over but that is not the case. The details
really matter in this sort of bugfixing, and I'm trying to get a clear
picture of what parts I may need to fix. *

Thanks

*Terry *

**Comment by Sikahalv** [comment]
Hi Terry,

I manually selected all the "correct" options I wanted, prior to creating the Support Zip File.

Now I've checked the settings again and they're all different and the app has a habit of forgetting each login... PLUS it's now changed different settings on it's own.

Also, regarding all of your steps, that's what I had already been doing the past few weeks -- prior to reaching out to you for assistance.

I've now uploaded a new Support Zip File showing the incorrect options that the app has selected on its own. For some reason the app has decided to *untick* the two options that I DO want enabled which are "Start automatically..." and "Start minimised"... but I don't want to see the splash screen on startup...

![Capture](https://user-images.githubusercontent.com/96361352/170397474-1b25eb99-2e01-4c88-889a-b3229caf104a.PNG)

[DisplayMagician-Support-20220526-1101.zip](https://github.com/terrymacdonald/DisplayMagician/files/8775553/DisplayMagician-Support-20220526-1101.zip)

**Comment by terrymacdonald** [comment]
@Sikahalv,

Some questions I have after looking at your original notes again:

   - Does DisplayMagician start at all originally after a computer restart
   at all?
   - If it does start originally, does it start minimised?
   - If you start DisplayMagician manually, does it start minimised?
   - If you start DisplayMagician manually, does it show the splash screen?
   - If you go into your Settings window, is the *Start DisplayMagician
   automatically when the computer starts *option currently enabled?
   - If you go into your Settings window, is the *Start DisplayMagician
   minimised in notification area *option currently enabled?
   - Do you run any anti-virus or other start menu adjusting software? Or
   any sort of software that prevents boot up of software?


I need those answers so that I can figure out which bit of code is not working properly.

Thanks
Terry

**Comment by terrymacdonald** [comment]
@Sikahalv When working on the code for issue #110 I found it is likely to have caused the issue you have experienced. The issue was related to the way that file upgrades have been performed in the past. I incorrectly upgrade the ProgramSettings to the wrong filename. This means that when DisplayMagican looks for the program settings file, it can't find it (as it's looking for the wrong name). This in turn means that it creates a new programsettings file each time.

This has been corrected, and hopefully should stop being an issue once DisplayMagician v2.3.2 is released.

I've also added some code to try and correct the wrong filename, so hopefully things start working the moment you upgrade. DisplayMagician v2.3.2 should be ready for release in about 2 weeks.

Thanks
Terry
